error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/StepVerifierDuplicateExpectNext.java

```diff
@@ -0,0 +1,115 @@
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+import reactor.test.StepVerifier;
+
+/**
+ * A {@link BugChecker} that flags duplicated usages of {@link StepVerifier.Step#expectNext} in
+ * favor of the overloaded variant.
+ *
+ * <p>Chaining {@link StepVerifier.Step#expectNext} calls can make the code more verbose than it has
+ * to be. Since {@link StepVerifier.Step#expectNext} offers several overload functions, those should
+ * be preferred over chaining {@link StepVerifier.Step#expectNext} calls.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary =
+        "When chaining multiple `StepVerifier.Step#expectNext` calls, please use the varargs overload instead",
+    link = BUG_PATTERNS_BASE_URL + "StepVerifierDuplicateExpectNext",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = SIMPLIFICATION)
+public final class StepVerifierDuplicateExpectNext extends BugChecker
+    implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Supplier<Type> STEP =
+      Suppliers.typeFromString("reactor.test.StepVerifier.Step");
+  private static final Matcher<ExpressionTree> STEP_EXPECTNEXT =
+      instanceMethod().onDescendantOf(STEP).named("expectNext");
+
+  /** Instantiates a new {@link StepVerifierDuplicateExpectNext} instance. */
+  public StepVerifierDuplicateExpectNext() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    // If the parent matches, this node will be considered when the parent parses its children, so
+    // we don't match it.
+    if (!STEP_EXPECTNEXT.matches(tree, state)
+        || getParent(tree).filter(t -> STEP_EXPECTNEXT.matches(t, state)).isPresent()) {
+      return Description.NO_MATCH;
+    }
+
+    @Var MethodInvocationTree child = tree;
+    List<ExpressionTree> newArgs = new ArrayList<>();
+
+    // The nodes are organized as MethodInvocationTree -> MemberSelectTree -> MethodInvocationTree
+    // We skip 2 to find the next method call in the call chain.
+    for (int nodeIndex = 2;
+        getChild(state, nodeIndex).filter(t -> STEP_EXPECTNEXT.matches(t, state)).isPresent();
+        nodeIndex += 2) {
+      // We checked in the loop condition that the child is present, so this is safe
+      child = getChild(state, nodeIndex).orElseThrow();
+      newArgs.addAll(child.getArguments());
+    }
+
+    if (newArgs.isEmpty()) {
+      return Description.NO_MATCH;
+    }
+
+    String newArgument = newArgs.stream().map(Object::toString).collect(joining(", "));
+    List<? extends ExpressionTree> myArgs = tree.getArguments();
+    SuggestedFix.Builder argumentsFix =
```

Comment:
`matchMethodInvocation` method is maybe a bit too full. Already made a comment about the middle part. Maybe we can extract the `Description` related things to a method like we do in some other places. Look for the regex: `try.*\(` to see some methods that return either a `Fix` or `Description` for some examples.